release notes july 16th 2024 (#4424)
Azure/AKS: CHANGELOG.md

# Azure Kubernetes Service Changelog

## Release 2024-07-16

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/). This release is titled as v20240716.

### Announcements

* 1.30 is the next LTS version after 1.27. Upgrade from 1.27 LTS to 1.30 LTS will be possible starting August 2024. More information about AKS LTS is available [here](https://learn.microsoft.com/azure/aks/long-term-support).

> * 1.27 community supported has ended. To exit [Long-Term Support (LTS)](https://learn.microsoft.com/azure/aks/long-term-support) upgrade to 1.28 and disable LTS with the `az aks upgrade` command.

Reviewer: nit/question: do you need to disable LTS before upgrade? Will you be charged if you upgrade and don't disable? Is there validation to prevent and ordered process? 

Not blocking as docs should have this clarified more than release notes

Author: @chasewilson - will take note of this and try and add it to docs. I don't know if you will be charged for this.

> * 1.27 community supported has ended. The ability to exit [Long-Term Support (LTS)](https://learn.microsoft.com/azure/aks/long-term-support) on 1.27 is blocked. To exit, upgrade to 1.28 and disable LTS with the `az aks upgrade` command.

Reviewer 2: 'The ability to exit [Long-Term Support (LTS)](https://learn.microsoft.com/azure/aks/long-term-support) on 1.27 is blocked' sounds a bit ominous whereas the line right after on upgrading to 1.28 and disabling sounds more positive. Can we reframe this entire point?

### Release Notes

* Features:
  * AKS version 1.30 is now GA.
  * AKS patch versions 1.30.2, 1.30.1, 1.29.6, 1.28.11, 1.27.15, are now available. Refer to [version support policy](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#kubernetes-version-support-policy) and [upgrading a cluster](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli) for more information.
  * [Istio add-on for AKS](https://learn.microsoft.com/azure/aks/istio-about) now supports EnvoyFilter of the type Lua `(type.googleapis.com/envoy.extensions.filters.http.lua.v3.Lua)`. While this `EnvoyFilter` is allowed, any issue arising from the Lua script itself is not supported. Other `EnvoyFilter` types currently remain blocked.
  * The ability to migrate your existing Ubuntu node pools to Azure Linux by changing the [OS SKU of the node pool](https://learn.microsoft.com/azure/azure-linux/tutorial-azure-linux-migration?tabs=azure-cli#in-place-os-sku-migration-preview) is now GA.

> * Preview feature:

Reviewer: ```suggestion
* Preview features:
```

>   * CNI Overlay dual-stack (IPv4/IPv6) is now available on Windows Agent Pools.

Reviewer 3: ```suggestion
  * CNI Overlay dual-stack (IPv4/IPv6) is now available on Windows Agent Pools.
  * Existing node pools can now be updated to enable or disable Federal Information Process Standard (FIPS). See [aka.ms/aks/updatefips](https://aka.ms/aks/updatefips) for more information.
```

Reviewer 3: New preview feature, LRB approved today, docs changes merging now

Author: ty!

  * CNI Overlay dual-stack (IPv4/IPv6) is now available on Windows Agent Pools.
  * Existing node pools can now be updated to enable or disable Federal Information Process Standard (FIPS). See [aka.ms/aks/updatefips](https://aka.ms/aks/updatefips) for more information.
  
* Bug Fixes:
  * Updated iptables rules in clusters with Azure Network Policy Manager to block pod access to wireserver.
  * A bug regarding App Routing's [placeholderPod](https://github.com/Azure/aks-app-routing-operator/blob/main/CHANGELOG.md#023-patch-1---2024-07-10) not properly cleaning up has been fixed.

> * Behavior Change

Reviewer: ```suggestion
* Behavior Change:
```

> * Component Updates:

Reviewer 3: Any links for keda or istio add-on release notes?

Author: Good call, I will link to KEDA. We actually have a brand new doc that came out too, so I will make sure to include that!

* Starting 1.30 Kubernetes version and 1.27 LTS versions, beta APIs will be disabled by default, when you upgrade to [them](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli#before-you-begin). There will be an option provided to explicitly enable beta APIs closer to the 1.30 release.
* 1.30 is the next LTS version after 1.27. Upgrade from 1.27 LTS to 1.30 LTS will be possible starting August 2024. More information about AKS LTS is available [here](https://learn.microsoft.com/azure/aks/long-term-support).
* 1.27 community supported has ended. The ability to exit [Long-Term Support (LTS)](https://learn.microsoft.com/azure/aks/long-term-support) on 1.27 is blocked. To exit, upgrade to 1.28 and disable LTS with the `az aks upgrade` command.

### Release Notes

* Features:
  * AKS version 1.30 is now GA.
  * AKS patch versions 1.30.2, 1.30.1, 1.29.6, 1.28.11, 1.27.15, are now available. Refer to [version support policy](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#kubernetes-version-support-policy) and [upgrading a cluster](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli) for more information.

>   * Under [Istio add-on](https://learn.microsoft.com/azure/aks/istio-about) for AKS, the add-on now supports EnvoyFilter of the type Lua `(type.googleapis.com/envoy.extensions.filters.http.lua.v3.Lua)`. While this `EnvoyFilter` is allowed, any issue arising from the Lua script itself is not supported. Other `EnvoyFilter` types currently remain blocked.

Reviewer 2: ```suggestion
  * [Istio add-on for AKS](https://learn.microsoft.com/azure/aks/istio-about) now supports EnvoyFilter of the type Lua `(type.googleapis.com/envoy.extensions.filters.http.lua.v3.Lua)`. While this `EnvoyFilter` is allowed, any issue arising from the Lua script itself is not supported. Other `EnvoyFilter` types currently remain blocked.
```

# Azure Kubernetes Service Changelog

## Release 2024-07-16

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/). This release is titled as v20240716.

### Announcements

> * Starting 1.30 Kubernetes version and 1.27 LTS versions, beta APIs will be disabled by default, when you upgrade to [them](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli#before-you-begin). There will be an option provided to explicitly enable beta APIs closer to the 1.30 release.

Reviewer 3: This release includes the GA of 1.30. Do we have the option for enabling beta APIs released?

* 1.30 is the next LTS version after 1.27. Upgrade from 1.27 LTS to 1.30 LTS will be possible starting August 2024. More information about AKS LTS is available [here](https://learn.microsoft.com/azure/aks/long-term-support).
* 1.27 community supported has ended. To exit [Long-Term Support (LTS)](https://learn.microsoft.com/azure/aks/long-term-support) upgrade to 1.28 and disable LTS with the `az aks upgrade` command.

### Release Notes

* Features:
  * AKS version 1.30 is now GA.
  * AKS patch versions 1.30.2, 1.30.1, 1.29.6, 1.28.11, 1.27.15, are now available. Refer to [version support policy](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#kubernetes-version-support-policy) and [upgrading a cluster](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli) for more information.
  * [Istio add-on for AKS](https://learn.microsoft.com/azure/aks/istio-about) now supports EnvoyFilter of the type Lua `(type.googleapis.com/envoy.extensions.filters.http.lua.v3.Lua)`. While this `EnvoyFilter` is allowed, any issue arising from the Lua script itself is not supported. Other `EnvoyFilter` types currently remain blocked.

>   * A new tutorial has been released on how to [securely scale your applications using the KEDA add-on and workload identity](https://learn.microsoft.com/azure/aks/keda-workload-identity).

Reviewer 2: this needs to be under announcements as it's not a new feature and instead a blog/announcement about existing feature

  * The ability to migrate your existing Ubuntu node pools to Azure Linux by changing the [OS SKU of the node pool](https://learn.microsoft.com/azure/azure-linux/tutorial-azure-linux-migration?tabs=azure-cli#in-place-os-sku-migration-preview) is now GA.

* Preview feature:
  * CNI Overlay dual-stack (IPv4/IPv6) is now available on Windows Agent Pools.
  * Existing node pools can now be updated to enable or disable Federal Information Process Standard (FIPS). See [aka.ms/aks/updatefips](https://aka.ms/aks/updatefips) for more information.
  
* Bug Fixes:

>   * Update iptables rules in clusters with Azure NetworkPolicy Manager to block pod access to wireserver.

Reviewer 2: ```suggestion
  * Updated iptables rules in clusters with Azure Network Policy Manager to block pod access to wireserver.
```

>   * [Cluster Autoscaler](https://github.com/kubernetes/autoscaler) has released patch v1.29.5 and v1.30.4 to fix a bug which prevents scaling from zero of selected SKUs.

Reviewer 2: this should be under Component updates with links to each of the releases IMO

  * A bug regarding App Routing's [placeholderPod](https://github.com/Azure/aks-app-routing-operator/blob/main/CHANGELOG.md#023-patch-1---2024-07-10) not properly cleaning up has been fixed.

* Behavior Change
  * AKS Automatic clusters now use Azure Linux for Node Auto Provision dynamic nodes.
     
* Component Updates:

>   * [Cluster Autoscaler](https://github.com/kubernetes/autoscaler) patches have been bumped to v1.29.5 and v1.30.4.

Reviewer 2: this point is repeated under bug fixes and component updates sections right now

  * coreDNS has been updated to use image  v1.9.4-hotfix.20240704 to fix CVE vulnerabilities.

>   * [Istio add-on](https://learn.microsoft.com/azure/aks/istio-about) has been bumped to include v1.21.3-hotfix.20240626 and v1.22.1-hotfix.20240626 to fix datadog tracer zero-day crash.

Reviewer 2: we don't officially support datadog tracer. I think this is a fix for Envoy AFAIK. @biefy can you share the list CVE numbers instead? @qpetraroia let's link to those CVEs instead of mentioning datadog tracer, that's better and also more informative IMO

Reviewer 4: Here is the CVE page: https://github.com/envoyproxy/envoy/security/advisories/GHSA-8mq4-c2v5-3h39
Yeah, we allow users to config the `DatadogTracingProvider`, but not supported.

>   * [KEDA add-on](https://learn.microsoft.com/azure/aks/keda-about) has been updated to v2.14.0 on AKS v1.30.

Reviewer 2: ```suggestion
  * [KEDA add-on](https://learn.microsoft.com/azure/aks/keda-about) has been updated to v2.14.0 on AKS v1.30.
```
```suggestion
  * [KEDA add-on](https://learn.microsoft.com/azure/aks/keda-about) has been updated to [v2.14.0](https://github.com/kedacore/keda/releases/tag/v2.14.0) on AKS v1.30.
```
